trieste/models/gpflow/sampler.py

```diff
@@ -512,22 +519,31 @@ def __init__(
         :param num_features: The number of features used to approximate the kernel. We use a default
             of 1000 as it typically perfoms well for a wide range of kernels. Note that very smooth
             kernels (e.g. RBF) can be well-approximated with fewer features.
-        :raise ValueError: If ``dataset`` is empty.
+        :raise NotImplementedError: If the model is not of valid type, or we try to use the
+            sampler with a model that has more than one latent GP.
         """
+        if model.model.num_latent_gps > 1:
+            raise NotImplementedError(
+                f"""
+                DecoupledTrajectorySampler does not currently support models with multiple latent
+                GPs, however received a model with {model.model.num_latent_gps} latent GPs.
+                """
+            )
```

Comment:
added a test that we are not getting a model with multiple latent gps, currently unsupported, added a line to the docstring

same thing added to both types of trajectory samplers